### PR TITLE
Add DH groups from RFC 7919 to support MySQL 8.1

### PR DESCRIPTION
**Pull request by samuel40791765**
### Issues:
Resolves `V1073444663`, `CryptoAlg-2101`

### Description of changes: 
MySQL's new version release depends on some new DH groups. This adds the support for the new group and updates our CI to test against the new version.

### Call-outs:
N/A

### Testing:
Had to disable two tests. One was complaining about session caching, one was complaining about supervisor processes (both cases are expected to fail).
* `main.client_ssl_data_print`
```
[ 69%] main.client_ssl_data_print               w7  [ retry-fail ]
        Test ended at 2023-12-21 01:57:34

CURRENT_TEST: main.client_ssl_data_print
--- /home/ubuntu/workplace/MYSQL_BUILD_ROOT/mysql-server/mysql-test/r/client_ssl_data_print.result	2023-12-21 03:46:37.812294083 +0300
+++ /home/ubuntu/workplace/MYSQL_BUILD_ROOT/mysql-aws-lc/mysql-test/var/7/log/client_ssl_data_print.reject	2023-12-21 04:57:34.726210533 +0300
@@ -1,3 +1,4 @@
 #
 # Bug#35304195: ssl_session_data_print creates world readable files by default
 #
+ERROR 2073 (HY000) at line 1: Can't get session data: session returned not resumable
``` 
* `main.udf_bug35242734`
```
main.udf_bug35242734                     w4  [ fail ]
        Test ended at 2023-12-21 02:06:23

CURRENT_TEST: main.udf_bug35242734
WARNING: MYSQL_OPT_RECONNECT is deprecated and will be removed in a future version.
mysqltest: At line 54: Server failed to restart
In included file ./include/wait_until_connected_again.inc: 57
included from /home/ubuntu/workplace/MYSQL_BUILD_ROOT/mysql-server/mysql-test/t/udf_bug35242734.test: 62

The result from queries just before the failure was:
# Verifying RESTART using mysqld_safe as supervisor process.
safe_process[125481]: Child process: 125482, exit: 1

 - the logfile can be found in '/home/ubuntu/workplace/MYSQL_BUILD_ROOT/mysql-aws-lc/mysql-test/var/log/main.udf_bug35242734/udf_bug35242734.log'
``` 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license and the ISC license.

### CODE REVIEW

File: tests/ci/integration/run_mysql_integration.sh [hunk 1]
```diff
@@ -61,6 +61,7 @@ function mysql_run_tests() {
   # currently support this.
   echo "main.mysqlpump_bugs : Bug#0000 Can't create/open a file ~/dump.sql'
 main.restart_server : Bug#0000 mysqld is not managed by supervisor process
+main.udf_bug35242734 : Bug#0000 mysqld is not managed by supervisor process
```
Comment (WillChilds-Klein) [inline tests/ci/integration/run_mysql_integration.sh:64]: is L59 still true?

File: crypto/dh_extra/dh_test.cc [hunk 2]
```diff
@@ -442,11 +566,13 @@ TEST(DHTest, RFC7919) {
 
 TEST(DHExpectedTestnputTest, CalculateSharedSecretMatches) {
   // KAT calculated with the following sage math code:
-  // prime=int("0x[prime for 2048 or 4096]", 16) R=Integers(prime) g = R(2)
-  // client_sk = int("0xABCDEF1234567890ABCDEF1234567890ABCDEF1234567890ABCDEF1234567890", 16) client_pk = g^client_sk
-  // server_sk = int("0xAABBCCDDEEFF11223344556677889900AABBCCDDEEFF11223344556677889900", 16) shared_secret = client_pk^server_sk
-  // print("client_pk", format(int(client_pk), '#x'))
-  // print("server_sk", format(server_sk, '#x'))
+  // prime=int("0x[prime for 2048 - 8192]", 16) R=Integers(prime) g = R(2)
```
Comment (WillChilds-Klein) [inline crypto/dh_extra/dh_test.cc:569]: suggest enumerating the field sizes and adding in semicolons to separate expressions

```suggestion
// prime=int("0x[prime for field sizes 2048, 3072, 4096, 8192", 16); R=Integers(prime); g = R(2)
```

File: crypto/dh_extra/dh_test.cc [hunk 3]
```diff
@@ -442,11 +566,13 @@ TEST(DHTest, RFC7919) {
...
+  // client_sk =
+  // int("0xABCDEF1234567890ABCDEF1234567890ABCDEF1234567890ABCDEF1234567890",
+  // 16) client_pk = g^client_sk server_sk =
+  // int("0xAABBCCDDEEFF11223344556677889900AABBCCDDEEFF11223344556677889900",
```
Comment (WillChilds-Klein) [inline crypto/dh_extra/dh_test.cc:573]: looks like clang-format messed up some whitespace-sensitive syntax?

**Review by WillChilds-Klein — COMMENTED** [review]
Looks like most of this diff is formatting changes. To confirm my understanding of this change, the only behavioral changes are the following?

1. Add 2 new security levels for FFDHE
2. Update test vectors, NIDs, etc. to accommodate ibid.
3. Upgrade MySQL integration version, skipping 2 additional unrelated test cases

Also, it looks like the new FFDHE implementations live inside the FIPS module boundary. Do we need to update the CAVP tooling or other FIPS things?

File: crypto/dh_extra/dh_test.cc (hunk 3 above)
Comment (samuel40791765) [reply crypto/dh_extra/dh_test.cc]: Ah yes, thanks for noticing!

**Comment by samuel40791765** [comment]
> Looks like most of this diff is formatting changes. To confirm my understanding of this change, the only behavioral changes are the following?

Yes these are the list of main changes. The NID files were autogenerated by running [`go run objects.go`](https://github.com/aws/aws-lc/blob/36c52853eda749c8835134f212ce90f47e7e1817/crypto/obj/README). I'll add the list to the commit message.

> Also, it looks like the new FFDHE implementations live inside the FIPS module boundary. Do we need to update the CAVP tooling or other FIPS things?

I don't think we've validated DH for FIPS. We have [tests](https://github.com/aws/aws-lc/blob/main/crypto/fipsmodule/service_indicator/service_indicator_test.cc#L4227-L4239) that mark DH as non-approved. We could expand upon those tests, but I think that deserves it's own PR since there are a few other groups we could additionally test for as well.

**Review by WillChilds-Klein — DISMISSED** [review]

**Review by andrewhop — APPROVED** [review]

**Review by WillChilds-Klein — APPROVED** [review]

**Pull request merged**